number_theory/numeric_unittest.cpp

```diff
@@ -0,0 +1,46 @@
+#include <stdint.h>
+
+#include <cmath>
+#include <limits>
+#include <random>
+
+#include <gtest/gtest.h>
+
+#include "number_theory/numeric.h"
+
+namespace tql {
+namespace number_theory {
+
+template <class T>
+void test_exgcd(T a, T b) {
+  auto [x, y] = exgcd(a, b);
+  if (a != 0 && b != 0) {
+    // Bézout's identity
+    ASSERT_LE(std::abs(x), std::abs(b));
+    ASSERT_LE(std::abs(y), std::abs(a));
+  }
+  ASSERT_EQ(x * a + y * b, gcd(a, b));
```

Comment:
I suggest using `std::gcd` explicitly in our own code so that we are aware that we are using the one from STL. Otherwise, it looks like its our own implementation which might require its own testing etc.